chainermn/communicators/pure_nccl_communicator.py

```diff
@@ -62,76 +65,113 @@ def _init_comms(self):
     def broadcast_data(self, model):
         _communication_utility.broadcast_naive(self.mpi_comm, model)
 
-    def allreduce_grad(self, model, stream=None):
-        self._init_comms()
-        if stream is None:
-            stream = chainer.cuda.Stream.null
+    def allreduce_grad(self, model):
+        stream = chainer.cuda.Stream.null
+        self.allreduce_grad_async(model, stream)
 
+    def allreduce_grad_async(self, model, stream):
```

Comment:
This clean up is great, but as this method is only for `_DoubleBufferingOptimizer` it should be private method, like named as `def _allreduce_grad_async(...)`.

Reply:
Ok. I will rename it.